src/ground-truth.rkt

```diff
@@ -53,17 +53,19 @@
         'unsamplable)
        y))))
 
-(define (eval-prog-real prog ctx)
-  (define repr (context-repr ctx))
-  (define pre `(λ ,(program-variables prog) (TRUE)))
-  (define fn (make-search-func pre (list prog) ctx))
+; ENSURE: all contexts have the same list of variables
+(define (eval-prog-list-real prog-list ctx-list)
```

Comment:
I would call this function `eval-progs-real` (following the convention that for a list we add an extra `s` to the end)

Reply:
Likewise for the variables.